File: functions/connect.js
const AWS = require('aws-sdk');

const table = 'your-table-name';
const dynamo = new AWS.DynamoDB.DocumentClient();

const handler = async (event) => {
  if (!event.requestContext) {
    return {
      statusCode: 400,
      body: JSON.stringify({
        message: 'Nothing to work with',
      }),
    };
  }

  const connectionId = event.requestContext.connectionId;

  await dynamo
    .put({
      TableName: table,
      Item: {
        connectionId: connectionId,
      },
    })
    .promise();

  return {
    statusCode: 200,
    body: JSON.stringify({ message: 'Connected succesfully.' }),
  };
};

module.exports = { handler };

File: functions/connect.py
import os, json, boto3

dynamodb = boto3.resource("dynamodb")
table = dynamodb.Table(os.getenv("TABLE_NAME"))


def handler(event, _):
    connection_id = event["requestContext"]["connectionId"]
    table.put_item(Item={"connectionId": connection_id})

    return {
        "statusCode": 200,
        "body": json.dumps({"message": "Connected successfully."}),
    }
